### PR TITLE
Use dismissible keyboard in acceptance flow

### PR DESCRIPTION
**Pull request by novalegra**
Update so the dismissible keyboard that Michael created is used in acceptance flow

### CODE REVIEW

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift [hunk 1]
```diff
@@ -84,16 +89,18 @@ struct PrescriptionCodeEntryView: View, HorizontalSizeClassOverride {
...
-        TextField(LocalizedString("Activation code", comment: "Placeholder text before entering prescription code in text field"), text: $prescriptionCode)
-        .keyboardType(.default)
-        .disableAutocorrection(true)
-        .font(.body)
-        .multilineTextAlignment(.leading)
+        DismissibleKeyboardTextField(
+            text: $prescriptionCode,
+            placeholder: LocalizedString("Activation code", comment: "Placeholder text before entering prescription code in text field"),
+            font: .preferredFont(forTextStyle: .body),
+            textColor: .darkGray,
```
Comment (mpangburn) [inline TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: Making sure this one is intended too: `.darkGray` is going to give you a different color than before; removing the argument will keep the default.

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift [hunk 2]
```diff
@@ -26,7 +27,11 @@ struct PrescriptionCodeEntryView: View, HorizontalSizeClassOverride {
             requestPrescriptionButton
             Spacer()
         }
-        .adaptiveKeyboardPadding() // To ensure the keyboard doesn't obstruct the TextField
+        .onKeyboardStateChange { state in
+            self.isKeyboardVisible = state.height > 0
+        }
+        .keyboardAware()
+        .edgesIgnoringSafeArea(isKeyboardVisible ? [] : .bottom)
```
Comment (mpangburn) [inline TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: Just making sure this is deliberate—your view now ignores the bottom safe area when the keyboard isn't visible, which is different behavior from before. (If you don't need this, you should be able to just use `keyboardAware` alone)

**Review by mpangburn — COMMENTED** [review]
Just checking in that a couple changes were intended

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift (hunk 2 above)
Comment (novalegra) [reply TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: Removed

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift (hunk 1 above)
Comment (novalegra) [reply TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: Thanks for the catch.

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift [hunk 3]
```diff
@@ -14,6 +14,7 @@ struct PrescriptionCodeEntryView: View, HorizontalSizeClassOverride {
     
     @State private var prescriptionCode: String = ""
     @ObservedObject var viewModel: PrescriptionReviewViewModel
+    @State private var isKeyboardVisible = false
```
Comment (mpangburn) [inline TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: I think this is unused now, right?

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift [hunk 4]
```diff
@@ -26,7 +27,10 @@ struct PrescriptionCodeEntryView: View, HorizontalSizeClassOverride {
             requestPrescriptionButton
             Spacer()
         }
-        .adaptiveKeyboardPadding() // To ensure the keyboard doesn't obstruct the TextField
+        .onKeyboardStateChange { state in
+            self.isKeyboardVisible = state.height > 0
+        }
```
Comment (mpangburn) [inline TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: as is this

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift [hunk 5]
```diff
@@ -84,16 +88,17 @@ struct PrescriptionCodeEntryView: View, HorizontalSizeClassOverride {
...
     private var prescriptionCodeInputField: some View {
-        TextField(LocalizedString("Activation code", comment: "Placeholder text before entering prescription code in text field"), text: $prescriptionCode)
-        .keyboardType(.default)
-        .disableAutocorrection(true)
-        .font(.body)
-        .multilineTextAlignment(.leading)
+        DismissibleKeyboardTextField(
+            text: $prescriptionCode,
+            placeholder: LocalizedString("Activation code", comment: "Placeholder text before entering prescription code in text field"),
+            font: .preferredFont(forTextStyle: .body),
```
Comment (mpangburn) [inline TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: Oh, and this `font` argument is the default, so you can remove it too

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift [hunk 6]
```diff
@@ -84,16 +88,17 @@ struct PrescriptionCodeEntryView: View, HorizontalSizeClassOverride {
...
+            textAlignment: .left,
```
Comment (mpangburn) [inline TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: The default argument here should suffice, you can remove this one

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift [hunk 7]
```diff
@@ -84,16 +88,17 @@ struct PrescriptionCodeEntryView: View, HorizontalSizeClassOverride {
...
+            keyboardType: .asciiCapable
```
Comment (mpangburn) [inline TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: And I suspect you can remove this one too, though it's worth double-checking

**Review by mpangburn — COMMENTED** [review]
I realized that your LoopKit PR put defaults in place for your new arguments that are specific to this use case—I made another comment on that one. Those defaults should match the UITextField defaults, since the wrapper is general purpose, and the specific autocapitalization/correction behavior you want should be determined at the call-site here.

File: TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift (hunk 7 above)
Comment (novalegra) [reply TidepoolServiceKitUI/Views/PrescriptionCodeEntryView.swift]: I added this so you can't enter emojis into the text field

**Review by mpangburn — APPROVED** [review]
Looks great!

**Pull request merged**